mage/query-modules/cpp/map.md

```diff
@@ -0,0 +1,351 @@
+---
+id: map
+title: map
+sidebar_label: map
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import RunOnSubgraph from '../../templates/_run_on_subgraph.mdx';
+
+export const Highlight = ({children, color}) => (
+<span
+style={{
+  backgroundColor: color,
+  borderRadius: '2px',
+  color: '#fff',
+  padding: '0.2rem',
+}}>
+{children}
+</span>
+);
+
+The map module offers a versatile toolkit for manipulating collections of key-value pairs, enabling advanced data operations within a graph database context
+
+[![docs-source](https://img.shields.io/badge/source-map-FB6E00?logo=github&style=for-the-badge)](https://github.com/memgraph/mage/tree/main/cpp/map_module)
+
+| Trait               | Value                                                 |
+| ------------------- | ----------------------------------------------------- |
+| **Module type**     | <Highlight color="#FB6E00">**algorithm**</Highlight>  |
+| **Implementation**  | <Highlight color="#FB6E00">**C++**</Highlight>        |
+| **Graph direction** | <Highlight color="#FB6E00">**directed**</Highlight>/<Highlight color="#FB6E00">**undirected**</Highlight> |
+| **Edge weights**    | <Highlight color="#FB6E00">**weighted**</Highlight>/<Highlight color="#FB6E00">**unweighted**</Highlight> |
+| **Parallelism**     | <Highlight color="#FB6E00">**sequential**</Highlight> |
+
+### Procedures
+
+### `remove_key(input_map, key, recursive_map)`
+
+Removes the specified key and its corresponding value from the input map. If the key does not exist in the input map, it will be ignored. Additionally, when the recursive setting is enabled, the key will also be removed from any inner maps that are part of the input map.
+
+#### Input:
+
+- `input_map: Map` ➡ the map from which the key is to be removed
+- `key: string` ➡ the key to be removed from the map
+- `recursive_map: bool` ➡ false by default, should be set to true if the input map consists of values that are also maps and, therefore, may have the key to be removed
+
+#### Output:
+
+- `removed: Map` ➡ the map after removing the specified key
+
+#### Usage:
+
+```cypher
+CALL map.remove_key({c: "b", d: "ba"}, "c") YIELD removed
+RETURN removed
+```
+
+```plaintext
++----------------------------+
+| removed                    |
++----------------------------+
+| {"d": "ba"}                |
++----------------------------+
+```
+
+Example with recursion:
+
+```cypher
+CALL map.remove_key({c: "b", d: {e: "ba", c: "h", a: {c: "z"}}}, "c", true) YIELD removed
+RETURN removed
+```
+
+```plaintext
++----------------------------+
+| removed                    |
++----------------------------+
+| {"d": {"e": "ba"}}         |
++----------------------------+
+```
+
+
+### `from_pairs(input_list)`
+
+Creates a map from a list of pairs, where each pair is essentially another list of size 2. The first element in each pair must be of type string, as it will be used as a key in the resulting map
```

Comment:
```suggestion
Creates a map from a list of pairs, where each pair is essentially another list of size 2. The first element in each pair must be of type `string`, as it will be used as a key in the resulting map.
```